src/modules/dataman/dataman.cpp

```diff
@@ -1098,13 +1092,16 @@ dm_write(dm_item_t item, unsigned index, dm_persitence_t persistence, const void
 	work->write_params.count = count;
 
 	/* Enqueue the item on the work queue and wait for the worker thread to complete processing it */
-	return (ssize_t)enqueue_work_item_and_wait_for_result(work);
+	ssize_t ret = (ssize_t)enqueue_work_item_and_wait_for_result(work);
+	perf_end(_dm_write_perf);
+	return ret;
 }
 
 /** Retrieve from the data manager file */
 __EXPORT ssize_t
 dm_read(dm_item_t item, unsigned index, void *buf, size_t count)
 {
+	perf_begin(_dm_read_perf);
```

Comment:
This is invalid if `!is_running()`. Can you move it down?

Reply:
Done.